### DISCUSSION TITLE
On the future of the plugin creator

### DISCUSSION THREAD

**Discussion by ecstrema** [musescore/MuseScore#16073] [Polls]
I was thinking about the plugin editor lately, and the more I think about it, the more I believe that an actual text editor has nothing to do in MuseScore. Other editors are far superior, provide richer mechanics and above all, are made for that. 

So, I'd see a simple and tiny window with this content: (please forget the emoji icons, it was the fastest option for a quick mockup):

![image](https://user-images.githubusercontent.com/35939574/214745647-cbec4a79-7424-49e7-9347-51bd2efa6916.png)

Completely remove any interface and keep only the console, where you should be able to see the output from all plugins individually.

You edit your file in a dedicated editor, like qtcreator, vscode, or any really. Then click reload, and run the plugin again.

Pros: Simple, no maintenance burden, easy workflow, let the IDE's do their job.
Cons: What about templates. Should there be a new button too? or a plugin that would be a plugin wizard? 

So I'm asking for your opinion: what do you think? Should the plugin creator keep the actual text editor, and try to integrate more features to make it more user friendly, or should it relegate that to IDEs and make it easier to view the actual plugin execution from within musescore.

**Comment by jeetee** [comment]
I'm somewhat ambivalent on this one, knowing that I often use Notepad++ at the moment to write the plugins.

Another (somewhat bigger) con of dropping the editor is having no way to test out not-yet-enabled/installed plugins. And then not knowing *why* they're not showing up because you can't get to their errors if you have no way to run them.

There was always some lingering idea of providing auto-completion inside the Plugin Creator and have the doxygen help included in it. To that end, I believe the effort would be pretty much the same to provide the meta-package for the MuseScore Plugin API to have it working in Qt Creator.

The biggest Pro of having that Qt Creator integration would mean we finally get the auto-completion and built-in help there whilst also benefitting from the Qt Designer stuff for checking UIs for the plugins during development.


So in summary:
I'd be fine with dropping the editor if we still have a way to run a chuck of code without the need to "install/enable" it and if the long-awaited improved support is then delivered by means of integration meta-package for Qt Creator.
